iot-hub/Samples/service/DigitalTwinClientSamples/Thermostat/ThermostatSample.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Serialization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Rest;
+using Newtonsoft.Json;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Samples
+{
+    public class ThermostatSample
+    {
+        private static readonly Random Random = new Random();
+        private readonly DigitalTwinClient _digitalTwinClient;
+        private readonly string _digitalTwinId;
+        private readonly ILogger _logger;
+
+        public ThermostatSample(DigitalTwinClient client, string digitalTwinId, ILogger logger)
+        {
+            _digitalTwinClient = client ?? throw new ArgumentNullException(nameof(client));
+            _digitalTwinId = digitalTwinId ?? throw new ArgumentNullException(nameof(digitalTwinId));
+            _logger = logger ?? LoggerFactory.Create(builder => builder.AddConsole()).CreateLogger<ThermostatSample>();
+        }
+
+        public async Task RunSampleAsync()
+        {
+            // Get and print the digital twin
+            await GetAndPrintDigitalTwin<ThermostatTwin>();
+
+            // Update the targetTemperature property of the digital twin
+            await UpdateTargetTemperatureProperty();
+
+            // Add, replace then remove currentTemperature property on the digital twin
+            await UpdateCurrentTemperatureProperty();
+
+            // Invoke the getMaxMinReport command on the root level of the digital twin
```

Comment:
nit: instead of saying - "invoke the command {abc} on the root level of the digital twin", how do you feel about saying "invoke the root-level command {abc} on the digital twin".

I think the 2nd one better highlights that we are invoking a root-level command.